libs/libc/stdlib/lib_strtof.c

```diff
@@ -69,28 +74,108 @@
 #  define __FLT_MAX_EXP__ (128)
 #endif
 
+#ifdef CONFIG_HAVE_LONG_DOUBLE
```

Comment:
if you convert all computation to long double, why not forward strtof to strod directly like musl?